homeassistant/components/zwave/migration.py

```diff
@@ -0,0 +1,166 @@
+"""Handle migration from legacy Z-Wave to OpenZWave and Z-Wave JS."""
+from __future__ import annotations
+
+from collections import defaultdict
+from typing import TYPE_CHECKING, TypedDict, cast
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import async_get as async_get_device_registry
+from homeassistant.helpers.entity_registry import async_get as async_get_entity_registry
+from homeassistant.helpers.singleton import singleton
+from homeassistant.helpers.storage import Store
+
+from .const import DOMAIN
+from .util import node_device_id_and_name
+
+if TYPE_CHECKING:
+    from . import ZWaveDeviceEntityValues
+
+LEGACY_ZWAVE_MIGRATION = f"{DOMAIN}_legacy_zwave_migration"
+STORAGE_WRITE_DELAY = 30
+STORAGE_KEY = f"{DOMAIN}.legacy_zwave_migration"
```

Comment:
Is it still needed to add `legacy_zwave` if we already prefix the value  with the `zwave` domain?

Reply:
Maybe not.

Reply:
One reason too keep it, is it'll be easier to identify the two storage files that are part of the same migration.